_posts/2018/2018-03-04-only-use-named-scopes-ouside-models.md

```diff
@@ -0,0 +1,71 @@
+---
+title: 'Only use named scopes outside models'
+description: 'Keep your wheres to yourself'
+layout: article
+category: ruby
+image:
+  base: '2018/only-use-named-scopes-outside-models'
+  alt: 'San Francisco Museum of Modern Art'
+  source: 'https://unsplash.com/photos/RkJF2BMrLJc'
+  credit: 'William Bout'
+---
+
+Last time we discussed using the hash-syntax in your where clauses but in my examples I did something
+
+## Instead of…
+
+...using `#where` scopes in your controllers or views.
+
+```ruby
+class PostsController < ApplicationController
+  def index
+    @posts = Post.where(status: 'published')
+  end
+end
+```
+
+
+## Use…
+
+...only named scopes that you define in your model.
+
+```ruby
+class Post < ApplicationRecord
+  scope :published, -> { where(status: 'published') }
+end
+```
+
+And then use it like:
+
+```ruby
+class PostsController < ApplicationController
+  def index
+    @posts = Post.published
+  end
+end
+```
+
+
+## But why?
+
+This technique is mostly to improve the organisation of your code. It forces you to do two things that can really help keep you productive over time.
+
+Firstly, naming the concepts you're creating. When you name something you explain it, often for the benefit of “future you”.
```

Comment:
*First: naming the concepts you're creating.

*"...future you" or your colleagues